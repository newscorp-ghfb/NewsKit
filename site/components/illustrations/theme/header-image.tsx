import React from 'react';
import {getSSRId} from 'newskit';
import {Path} from '../path';
import {Circle} from '../circle';
import {Rect} from '../rect';
import {Svg} from '../svg';

export const HeaderImage: React.FC = () => {
  const clip0 = getSSRId();
  const filter0 = getSSRId();
  const filter1 = getSSRId();
  const filter2 = getSSRId();
  const filter3 = getSSRId();

  return (
    <Svg
      width="1572"
      height="997"
      viewBox="0 0 1572 997"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g clipPath={`url(#${clip0})`}>
        <g filter={`url(#${filter0})`}>
          <Path
            d="M1180.21 414.822C1252.87 395.026 1331.12 449.976 1354.98 537.556C1378.84 625.137 1339.28 712.183 1266.62 731.979L439.793 938.658C367.132 958.454 463.654 874.474 439.793 786.893C415.932 699.313 455.493 612.267 528.154 592.471L1180.21 414.822Z"
            fill="white"
          />
        </g>
        <Circle
          cx="1196.92"
          cy="584.285"
          r="126.087"
          transform="rotate(14.76 1196.92 584.285)"
          fill="#7264EF"
          stroke="#E4E4E4"
          stroke-width="1.5724"
        />
        <Circle
          cx="920.831"
          cy="653.186"
          r="126.087"
          transform="rotate(14.76 920.831 653.186)"
          fill="#9B87FC"
          stroke="#E4E4E4"
          stroke-width="1.5724"
        />
        <Circle
          cx="644.01"
          cy="724.875"
          r="126.087"
          transform="rotate(14.76 644.01 724.875)"
          fill="#BBAEFF"
          stroke="#E4E4E4"
          stroke-width="1.5724"
        />
        <g filter={`url(#${filter1})`}>
          <Path
            d="M1050.78 223.091C1116.81 184.565 1207.43 216.894 1253.17 295.3C1298.91 373.706 1282.45 468.498 1216.41 507.024L485.905 919.789C419.866 958.315 467.126 858.949 421.386 780.543C375.646 702.137 392.102 607.345 458.142 568.819L1050.78 223.091Z"
            fill="white"
          />
        </g>
        <Circle
          cx="1118.23"
          cy="376.873"
          r="126.087"
          transform="rotate(-0.258284 1118.23 376.873)"
          fill="#65BBC4"
          stroke="#E4E4E4"
          stroke-width="1.5724"
        />
        <Circle
          cx="869.431"
          cy="514.961"
          r="126.087"
          transform="rotate(-0.258284 869.431 514.961)"
          fill="#81D2DA"
          stroke="#E4E4E4"
          stroke-width="1.5724"
        />
        <Circle
          cx="620.643"
          cy="655.933"
          r="126.087"
          transform="rotate(-0.258284 620.643 655.933)"
          fill="#A8E0E6"
          stroke="#E4E4E4"
          stroke-width="1.5724"
        />
        <g filter={`url(#${filter2})`}>
          <Path
            fillRule="evenodd"
            clipRule="evenodd"
            d="M1111.87 103.614C1049.04 38.1054 944.992 35.938 879.483 98.773L291.614 662.647C226.105 725.482 223.937 829.525 286.772 895.034C349.607 960.543 453.651 962.71 519.16 899.875L1107.03 336.002C1172.54 273.167 1174.71 169.123 1111.87 103.614ZM400.051 838.502C419.382 833.752 431.202 814.23 426.451 794.899C421.701 775.567 402.179 763.748 382.847 768.498C363.516 773.249 351.696 792.771 356.447 812.102C361.197 831.433 380.719 843.253 400.051 838.502Z"
            fill="white"
          />
        </g>
        <g filter={`url(#${filter3})`}>
          <Path
            fillRule="evenodd"
            clipRule="evenodd"
            d="M518.072 898.741L1105.94 334.867C1170.82 272.633 1172.97 169.585 1110.74 104.703C1048.5 39.8206 945.455 37.6739 880.572 99.9078L292.703 663.782C227.821 726.015 225.674 829.063 287.908 893.946C350.142 958.828 453.19 960.975 518.072 898.741ZM1111.87 103.614C1049.04 38.1055 944.993 35.938 879.484 98.773L291.615 662.647C226.106 725.482 223.938 829.525 286.773 895.034C349.608 960.543 453.652 962.71 519.161 899.875L1107.03 336.002C1172.54 273.167 1174.71 169.123 1111.87 103.614Z"
            fill="#E4E4E4"
          />
        </g>
        <Circle
          cx="995.394"
          cy="220.237"
          r="126.087"
          transform="rotate(-13.8065 995.394 220.237)"
          fill="#577FFB"
          stroke="#E4E4E4"
          stroke-width="1.5724"
        />
        <Circle
          cx="785.866"
          cy="412.767"
          r="126.087"
          transform="rotate(-13.8065 785.866 412.767)"
          fill="#87A4FC"
          stroke="#E4E4E4"
          stroke-width="1.5724"
        />
        <Circle
          cx="577.024"
          cy="608.098"
          r="126.087"
          transform="rotate(-13.8065 577.024 608.098)"
          fill="#AEBFFF"
          stroke="#E4E4E4"
          stroke-width="1.5724"
        />
      </g>
      <defs>
        <filter
          id={filter0}
          x="387.031"
          y="398.207"
          width="1000.05"
          height="581.114"
          filterUnits="userSpaceOnUse"
          colorInterpolationFilters="sRGB"
        >
          <feFlood floodOpacity="0" result="BackgroundImageFix" />
          <feColorMatrix
            in="SourceAlpha"
            type="matrix"
            values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
            result="hardAlpha"
          />
          <feOffset dy="12.5792" />
          <feGaussianBlur stdDeviation="12.5792" />
          <feColorMatrix
            type="matrix"
            values="0 0 0 0 0.0392157 0 0 0 0 0.0392157 0 0 0 0 0.0392157 0 0 0 0.08 0"
          />
          <feBlend
            mode="normal"
            in2="BackgroundImageFix"
            result="effect1_dropShadow_751_135603"
          />
          <feBlend
            mode="normal"
            in="SourceGraphic"
            in2="effect1_dropShadow_751_135603"
            result="shape"
          />
        </filter>
        <filter
          id={filter1}
          x="370.334"
          y="194.264"
          width="933.889"
          height="771.832"
          filterUnits="userSpaceOnUse"
          colorInterpolationFilters="sRGB"
        >
          <feFlood floodOpacity="0" result="BackgroundImageFix" />
          <feColorMatrix
            in="SourceAlpha"
            type="matrix"
            values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
            result="hardAlpha"
          />
          <feOffset dy="12.5792" />
          <feGaussianBlur stdDeviation="12.5792" />
          <feColorMatrix
            type="matrix"
            values="0 0 0 0 0.0392157 0 0 0 0 0.0392157 0 0 0 0 0.0392157 0 0 0 0.08 0"
          />
          <feBlend
            mode="normal"
            in2="BackgroundImageFix"
            result="effect1_dropShadow_751_135603"
          />
          <feBlend
            mode="normal"
            in="SourceGraphic"
            in2="effect1_dropShadow_751_135603"
            result="shape"
          />
        </filter>
        <filter
          id={filter2}
          x="215.87"
          y="40.4496"
          width="966.904"
          height="942.907"
          filterUnits="userSpaceOnUse"
          colorInterpolationFilters="sRGB"
        >
          <feFlood floodOpacity="0" result="BackgroundImageFix" />
          <feColorMatrix
            in="SourceAlpha"
            type="matrix"
            values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
            result="hardAlpha"
          />
          <feOffset dy="12.5792" />
          <feGaussianBlur stdDeviation="12.5792" />
          <feColorMatrix
            type="matrix"
            values="0 0 0 0 0.0392157 0 0 0 0 0.0392157 0 0 0 0 0.0392157 0 0 0 0.08 0"
          />
          <feBlend
            mode="normal"
            in2="BackgroundImageFix"
            result="effect1_dropShadow_751_135603"
          />
          <feBlend
            mode="normal"
            in="SourceGraphic"
            in2="effect1_dropShadow_751_135603"
            result="shape"
          />
        </filter>
        <filter
          id={filter3}
          x="215.871"
          y="40.4496"
          width="966.903"
          height="942.907"
          filterUnits="userSpaceOnUse"
          colorInterpolationFilters="sRGB"
        >
          <feFlood floodOpacity="0" result="BackgroundImageFix" />
          <feColorMatrix
            in="SourceAlpha"
            type="matrix"
            values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
            result="hardAlpha"
          />
          <feOffset dy="12.5792" />
          <feGaussianBlur stdDeviation="12.5792" />
          <feColorMatrix
            type="matrix"
            values="0 0 0 0 0.0392157 0 0 0 0 0.0392157 0 0 0 0 0.0392157 0 0 0 0.08 0"
          />
          <feBlend
            mode="normal"
            in2="BackgroundImageFix"
            result="effect1_dropShadow_751_135603"
          />
          <feBlend
            mode="normal"
            in="SourceGraphic"
            in2="effect1_dropShadow_751_135603"
            result="shape"
          />
        </filter>
        <clipPath id={clip0}>
          <Rect width="1572" height="997" fill="white" />
        </clipPath>
      </defs>
    </Svg>
  );
};

export default HeaderImage;
