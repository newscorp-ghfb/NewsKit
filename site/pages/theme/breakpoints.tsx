import React from 'react';
import {newskitLightTheme, compileTheme} from 'newskit';
import {FoundationPageTemplate} from '../../templates/foundation-page-template';
import {ComponentPageCell} from '../../components/layout-cells';
import {
  ContentSection,
  ContentPrimary,
  ContentSecondary,
} from '../../components/content-structure';
import {LayoutProps} from '../../components/layout';
import {
  Illustration,
  getIllustrationComponent,
} from '../../components/illustrations/illustration-loader';
import {Table} from '../../components/table';
import {Code} from '../../components/code';
import {Link} from '../../components/link';
import {MediaList} from '../../components/media-list';
import {InlineCode} from '../../components/markdown-elements';

const compiledNewskitLightTheme = compileTheme(newskitLightTheme);

const RANGE_DESCRIPTION: {[key: string]: string} = {
  xs: '0-479',
  sm: '480-767',
  md: '768-1023',
  lg: '1024-1439',
  xl: '1440>',
};

const USAGE_DESCRIPTION: {[key: string]: string | JSX.Element} = {
  xs: 'Portrait - small handset >  large handset',
  sm: (
    <>
      Portrait - large handset &gt; small tablet
      <br />
      Landscape - small handset
    </>
  ),
  md: (
    <>
      Portrait - large tablet
      <br />
      Landscape - large handset &gt; small tablet
    </>
  ),
  lg: 'Landscape - large tablet > desktop',
  xl: 'Landscape - large tablet > desktop',
};

const bps = compiledNewskitLightTheme.breakpoints;

const breakpointsRows = Object.entries(bps).map(([tokenName, tokenValue]) => ({
  token: tokenName,
  breakpoint: tokenValue,
  range: RANGE_DESCRIPTION[tokenName] || '',
  usage: USAGE_DESCRIPTION[tokenName] || '',
}));

export default (layoutProps: LayoutProps) => (
  <FoundationPageTemplate
    headTags={{
      title: 'Breakpoints',
      description:
        'Breakpoints set a visual point on a screen to adapt the design of content (responsive design), ensuring a consistent, optimised experience across different screen widths.',
    }}
    layoutProps={layoutProps}
    pageIntroduction={{
      type: 'Foundations',
      name: 'Breakpoints',
      hero: {
        illustration: 'theme/breakpoints/hero',
      },
      introduction: `Breakpoints set a visual point on a screen to adapt the design of content (responsive design), ensuring a consistent, optimised experience across different screen widths.`,
    }}
  >
    <ComponentPageCell>
      <ContentSection sectionName="overview">
        <ContentPrimary
          id="overview"
          toc="Overview"
          headline="Overview"
          description={
            <>
              There are five breakpoints available and the breakpoint range,
              margins and gutters can be customised.
              <br />
              <br />
              The default values are outlined below:
            </>
          }
        >
          <Illustration path="theme/breakpoints/overview" />
        </ContentPrimary>

        <ContentSecondary>
          <Table
            columns={['Token', 'Breakpoint', 'Range', 'Usage']}
            rows={breakpointsRows}
          />
        </ContentSecondary>

        <ContentSecondary description="Default breakpoints can be overriden in the theme.">
          <Code>
            {`import {createTheme} from 'newskit'; 

const theme = createTheme({ 
    name: 'theme-with-custom-breakpoints', 
    overrides: { 
        breakpoints: { 
            xs: 0, // 0-359 
            sm: 360, 
            md: 540, 
            lg: 720, 
            xl: 1080, 
        } 
    }, 
});`}
          </Code>
        </ContentSecondary>

        <ContentSecondary description="All breakpoints are fluid until the maximum grid width of 1920px (default), at this point the grid becomes fixed.">
          <MediaList
            cardsLayout={{
              xs: 'vertical',
              sm: 'vertical',
              md: 'horizontal',
              lg: 'horizontal',
              xl: 'horizontal',
            }}
            layout="1-span"
            cards={[
              {
                title: 'Fluid',
                description: `
                  Fluid layouts are calculated to their relative size, and
                  stretch as the viewport is resized. For example, if you set
                  the width of your layout to 100% width, each column will only
                  be calculated to its relative size., and will stretch as the
                  browser is resized.
                  `,
                media: getIllustrationComponent('theme/breakpoints/fluid'),
              },
              {
                title: 'Fixed',
                description: `
                  Fixed layouts do not respond to the size of the screen (viewport) but remain at a fixed width at a specific numerical value e.g. 1920px.
                  `,
                media: getIllustrationComponent('theme/breakpoints/fixed'),
              },
            ]}
          />
        </ContentSecondary>

        <ContentSecondary
          description={
            <>
              Max-width defaults can be overidden in the{' '}
              <Link
                href="/components/grid/"
                overrides={{typographyPreset: 'editorialParagraph030'}}
              >
                Grid & Cell component.
              </Link>
            </>
          }
          showSeparator
        />
      </ContentSection>

      <ContentSection sectionName="usage">
        <ContentPrimary
          id="usage"
          toc="Using breakpoints"
          headline="Using breakpoints"
          description="NewsKit enables the appearance and behaviour of UI elements to change at different breakpoints."
        >
          <Illustration path="theme/breakpoints/usage" />
        </ContentPrimary>

        <ContentSecondary
          description={
            <>
              In the codebase, changing the appearance and behaviour of UI
              elements can be achieved by utilising the{' '}
              <Link href="/theme/theming/component-defaults/">
                component defaults and overrides system
              </Link>{' '}
              to override design tokens at specific breakpoints. For example,
              setting different typography presets at different breakpoints, can
              make the text larger on bigger devices than on smaller devices.
              <br />
              <br />
              Properties that support this are specified in the Props sections
              of the component documentation as using the generic{' '}
              <InlineCode>MQ</InlineCode> type, i.e.{' '}
              <InlineCode>{`MQ<string>`}</InlineCode>.
              <br />
              <br />
              This is done by passing a single string token to be used across
              all breakpoints, or an object for a combination of multiple
              breakpoints - comprising of breakpoint keys and token values.
            </>
          }
        />

        <ContentSecondary
          showSeparator
          headline="Code example"
          description="This example below shows different typography presets being used at different breakpoint."
        >
          <Code>{`
    typographyPreset: {
        xs: 'editorialHeadline010',
        md: 'editorialHeadline020',
        lg: 'editorialHeadline030',
      }
    `}</Code>
        </ContentSecondary>
      </ContentSection>
    </ComponentPageCell>
  </FoundationPageTemplate>
);
