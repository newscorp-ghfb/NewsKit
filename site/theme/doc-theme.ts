import {createTheme, newskitLightTheme} from 'newskit';
import documentationLight from '@newskit-themes/newskit-website/Documentation-light.json';
import documentationLightAccessibility from '@newskit-themes/newskit-website/Documentation-light-accessibility.json';
import documentationLightFoundations from '@newskit-themes/newskit-website/Documentation-light-foundations.json';
import documentationLightPatterns from '@newskit-themes/newskit-website/Documentation-light-patterns.json';

import documentationDark from '@newskit-themes/newskit-website/Documentation-dark.json';
import documentationDarkAccessibility from '@newskit-themes/newskit-website/Documentation-dark-accessibility.json';
import documentationDarkFoundations from '@newskit-themes/newskit-website/Documentation-dark-foundations.json';
import documentationDarkPatterns from '@newskit-themes/newskit-website/Documentation-dark-patterns.json';

import {stylePresets} from './style-presets';
import {componentDefaults} from './component-defaults';

// How we get font metrics in.
// If we had different fonts in different themes, this implementation would look different
// Docs site uses the same fonts for each theme - so we just use the same from Light theme as the 'default'
const {fonts} = documentationLight;

Object.assign(fonts.fontFamily010, {
  fontMetrics: {
    fontWeight010: {
      ascent: 992,
      capHeight: 700,
      descent: -310,
      lineGap: 0,
      unitsPerEm: 1000,
    },
  },
});
Object.assign(fonts.fontFamily020, {
  fontMetrics: {
    fontWeight010: {
      ascent: 992,
      capHeight: 700,
      descent: -310,
      lineGap: 0,
      unitsPerEm: 1000,
    },
  },
});
Object.assign(fonts.fontFamily030, {
  fontMetrics: {
    fontWeight010: {
      ascent: 992,
      capHeight: 700,
      descent: -310,
      lineGap: 0,
      unitsPerEm: 1000,
    },
  },
});

export const docsThemeLight = createTheme({
  name: 'docs-theme-light',
  baseTheme: newskitLightTheme,
  overrides: {
    ...documentationLight,
    fonts,
    stylePresets,
    componentDefaults,
  },
});

export const docsThemeDark = createTheme({
  name: 'docs-theme-dark',
  baseTheme: docsThemeLight,
  overrides: {
    ...documentationDark,
    fonts,
    outlines: {
      safariOutlineStyleDeafult: 'solid',
    },
  },
});

export const foundationsThemeLight = createTheme({
  name: 'docs-foundations-light',
  baseTheme: docsThemeLight,
  overrides: {...documentationLightFoundations, fonts},
});

export const foundationsThemeDark = createTheme({
  name: 'docs-foundations-dark',
  baseTheme: docsThemeDark,
  overrides: {...documentationDarkFoundations, fonts},
});

export const accessibilityThemeLight = createTheme({
  name: 'docs-accessibilits-light',
  baseTheme: docsThemeLight,
  overrides: {...documentationLightAccessibility, fonts},
});

export const accessibilitysThemeDark = createTheme({
  name: 'docs-accessibility-dark',
  baseTheme: docsThemeDark,
  overrides: {...documentationDarkAccessibility, fonts},
});

export const patternsThemeLight = createTheme({
  name: 'docs-patterns-light',
  baseTheme: docsThemeLight,
  overrides: {...documentationLightPatterns, fonts},
});

export const patternsThemeDark = createTheme({
  name: 'docs-patterns-dark',
  baseTheme: docsThemeDark,
  overrides: {...documentationDarkPatterns, fonts},
});

// These are used on guide pages â€“ initally were going to be another colour
// Now they are the same as the default theme

export const guidesThemeLight = createTheme({
  name: 'docs-guides-light',
  baseTheme: docsThemeLight,
});

export const guidesThemeDark = createTheme({
  name: 'docs-guides-dark',
  baseTheme: docsThemeDark,
});
