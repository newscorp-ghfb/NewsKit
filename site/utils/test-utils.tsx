/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme: any = renderToFragmentWithThemeFactory(
  docsThemeLight,
);

export const mockNewsKitComponents = (...components: string[]) => () => ({
  ...(jest.requireActual('newskit') as any),
  ...components.reduce((acc, comp) => {
    acc[comp] = ({children, ...props}: any) => (
      <div
        data-comp={`NewsKit ${comp}`}
        data-testid={comp}
        data-props={JSON.stringify(props, null, 2)}
      >
        {children}
      </div>
    );
    return acc;
  }, {} as Record<string, Function>),
});
