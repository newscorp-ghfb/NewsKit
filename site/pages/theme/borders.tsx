import React from 'react';
import {InlineMessage, newskitLightTheme, compileTheme} from 'newskit';
import {FoundationPageTemplate} from '../../templates/foundation-page-template';
import {LayoutProps} from '../../components/layout';
import {Table} from '../../components/table';
import {getTokenType} from '../../utils/get-token-type';
import {Code} from '../../components/code';
import {
  ContentSection,
  ContentPrimary,
  ContentSecondary,
} from '../../components/content-structure';
import {ComponentPageCell} from '../../components/layout-cells';

const compiledNewskitLightTheme = compileTheme(newskitLightTheme);
const borderRows = (type: string) =>
  getTokenType(compiledNewskitLightTheme.borders, type).map(
    ({tokenName, tokenValue}) => ({
      [type]: tokenName,
      token: tokenName,
      tokenValue: tokenValue as string,
    }),
  );

type CodeExampleType = {
  [key: string]: string;
};

const codeExamples: CodeExampleType = {
  borderRadius: `borderRadius: '{{borders.borderRadiusDefault}}'`,
  borderWidth: `borderWidth: '{{borders.borderWidth000}} {{borders.borderWidth000}} {{borders.borderWidth010}} {{borders.borderWidth000}}'`,
  borderColor: `borderColor: '{{colors.interactivePrimary030}}'`,
  borderStyle: `borderStyle: 'solid'`,
};

interface ExampleSectionProps {
  section: string;
  showSeparator?: boolean;
}

const ExampleSection: React.FC<ExampleSectionProps> = ({
  section,
  showSeparator,
}) => (
  <ContentSecondary headline="Code example" showSeparator={showSeparator}>
    <Code>{codeExamples[section]}</Code>
  </ContentSecondary>
);

export default (layoutProps: LayoutProps) => (
  <FoundationPageTemplate
    headTags={{
      title: 'Borders',
      description:
        'Borders direct attention, identify components, communicate state, and express a brand.',
    }}
    layoutProps={layoutProps}
    pageIntroduction={{
      type: 'Foundations',
      name: 'Borders',
      hero: {
        illustration: 'theme/hero-borders-illustration',
      },
      introduction: `Borders direct attention, identify components, communicate state, and express a brand.`,
    }}
  >
    <ComponentPageCell>
      <ContentSection sectionName="border-radius">
        <ContentPrimary
          id="border-radius"
          toc="Border Radius"
          headline="Border Radius"
          description="Applying a Border Radius to an element rounds its corners. NewsKit uses a simple set of design token to ensure consistency throughout the UI."
        >
          <Table
            columns={['Border radius', 'Token', 'Token value']}
            rows={borderRows('borderRadius')}
          />
        </ContentPrimary>

        <ExampleSection section="borderRadius" showSeparator />
      </ContentSection>

      <ContentSection sectionName="border-width">
        <ContentPrimary
          id="border-width"
          toc="Border Width"
          headline="Border Width"
          description="Applying a Border Width to an element defines the width of the element’s border. NewsKit uses a simple set of design token to ensure consistency throughout the UI."
        >
          <Table
            columns={['Border width', 'Token', 'Token value']}
            rows={borderRows('borderWidth')}
          />
        </ContentPrimary>

        <ExampleSection section="borderWidth" />

        <ContentSecondary showSeparator>
          <InlineMessage title="Note">
            In addition to the Border Radius and Border Width foundations,
            Border Colour and Border Style can be applied to a UI element using
            the borderColor and borderStyle attribute on a Style Preset.
          </InlineMessage>
        </ContentSecondary>
      </ContentSection>

      <ContentSection sectionName="border-colour">
        <ContentPrimary
          id="border-colour"
          toc="Border Colour"
          headline="Border Colour"
          description="Applying a Border Colour to an element defines the colour of the element's border. Border colour utilises Interface and Interactive colour tokens."
        />
        <ExampleSection section="borderColor" showSeparator />
      </ContentSection>

      <ContentSection sectionName="border-style">
        <ContentPrimary
          id="border-style"
          toc="Border Style"
          headline="Border Style"
          description="Applying a Border Style to an element defines the style of the element’s border. Border Style utilises the standard CSS Border Style values."
        />

        <ExampleSection section="borderStyle" showSeparator />
      </ContentSection>
    </ComponentPageCell>
  </FoundationPageTemplate>
);
