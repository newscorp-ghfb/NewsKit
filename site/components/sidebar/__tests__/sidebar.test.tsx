import {renderToFragmentInBody} from 'newskit/test/test-utils';
import {docsThemeLight} from '../../../theme/doc-theme';
import Sidebar from '..';

const useRouter = jest.spyOn(require('next/router'), 'useRouter');

useRouter.mockImplementation(() => ({pathname: '/'}));

describe('Sidebar', () => {
  describe('renders correctly', () => {
    test('when open', () => {
      const fragment = renderToFragmentInBody(
        Sidebar,
        {
          sidebarOpen: true,
          handleSidebarClick: () => {},
        },
        docsThemeLight,
      );
      expect(fragment).toMatchSnapshot();
    });

    test('when closed', () => {
      const fragment = renderToFragmentInBody(
        Sidebar,
        {
          sidebarOpen: false,
          handleSidebarClick: () => {},
        },
        docsThemeLight,
      );
      expect(fragment).toMatchSnapshot();
    });
  });
});
