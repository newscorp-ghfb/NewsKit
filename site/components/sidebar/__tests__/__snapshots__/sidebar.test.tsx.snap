// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders correctly when closed 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 276px;
  padding-left: 24px;
  padding-right: 24px;
  position: fixed;
  overflow: auto;
  bottom: 0;
  top: 0;
  left: 0;
  z-index: 3;
  -webkit-transform: translateX(-100%);
  -moz-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
  -webkit-transition: -webkit-transform 300ms;
  transition: transform 300ms;
  margin-top: 64px;
  background-color: #F4F4F4;
  border-style: solid;
  border-color: #E4E4E4;
  border-width: 0 1px 0 0;
}

@media screen and (min-width: 1024px) {
  .emotion-0 {
    margin-top: 80px;
    -webkit-transform: translateX(0);
    -moz-transform: translateX(0);
    -ms-transform: translateX(0);
    transform: translateX(0);
    overflow: hidden;
  }

  .emotion-0:hover {
    overflow: auto;
  }
}

@media only screen and (min-device-width: 1024px) and (max-device-width: 1366px) and (-webkit-min-device-pixel-ratio: 2) {
  .emotion-0 {
    overflow: auto;
  }
}

.emotion-1 {
  padding: 48px;
}

@media screen {
  .emotion-2 {
    display: block;
  }
}

@media screen and (min-width: 480px) {
  .emotion-2 {
    display: block;
  }
}

@media screen and (min-width: 768px) {
  .emotion-2 {
    display: block;
  }
}

@media screen and (min-width: 1024px) {
  .emotion-2 {
    display: none;
  }
}

@media screen and (min-width: 1440px) {
  .emotion-2 {
    display: none;
  }
}

.emotion-3 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-4 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.emotion-5 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
  fill: #2E2E2E;
  vertical-align: unset;
  display: inline-block;
}

.emotion-5.emotion-5 {
  width: 16px;
  height: 16px;
  fill: #2E2E2E;
}

.emotion-7 {
  font-family: "DM Sans",sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  letter-spacing: 0;
}

.emotion-8 {
  list-style-type: none;
  margin: 0px;
  padding: 0;
}

.emotion-8:last-child {
  margin-bottom: 24px;
}

<div
    class="emotion-0"
    data-testid="sidebar"
    role="complementary"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-3"
        >
          <div
            class="emotion-4"
            data-testid="close-icon"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="emotion-5 emotion-6"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <nav
      aria-label="Sidebar"
      class="emotion-7"
      role="navigation"
    >
      <ul
        class="emotion-8"
        role="list"
      />
    </nav>
  </div>
  .emotion-0 {
  background-color: rgba(0, 0, 0, 0.8);
  position: fixed;
  overflow: auto;
  bottom: 0;
  top: 0;
  left: 0;
  right: 100%;
  opacity: 0;
  visibility: hidden;
  -webkit-transition: opacity 0.6s,right 0.3s;
  transition: opacity 0.6s,right 0.3s;
  z-index: 2;
}

@media screen and (min-width: 1024px) {
  .emotion-0 {
    display: none;
  }
}

<div
    class="emotion-0"
    data-testid="overlay"
  />
</DocumentFragment>
`;

exports[`Sidebar renders correctly when open 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 276px;
  padding-left: 24px;
  padding-right: 24px;
  position: fixed;
  overflow: auto;
  bottom: 0;
  top: 0;
  left: 0;
  z-index: 3;
  -webkit-transform: translateX(0);
  -moz-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  -webkit-transition: -webkit-transform 300ms;
  transition: transform 300ms;
  margin-top: 64px;
  background-color: #F4F4F4;
  border-style: solid;
  border-color: #E4E4E4;
  border-width: 0 1px 0 0;
}

@media screen and (min-width: 1024px) {
  .emotion-0 {
    margin-top: 80px;
    -webkit-transform: translateX(0);
    -moz-transform: translateX(0);
    -ms-transform: translateX(0);
    transform: translateX(0);
    overflow: hidden;
  }

  .emotion-0:hover {
    overflow: auto;
  }
}

@media only screen and (min-device-width: 1024px) and (max-device-width: 1366px) and (-webkit-min-device-pixel-ratio: 2) {
  .emotion-0 {
    overflow: auto;
  }
}

.emotion-1 {
  padding: 48px;
}

@media screen {
  .emotion-2 {
    display: block;
  }
}

@media screen and (min-width: 480px) {
  .emotion-2 {
    display: block;
  }
}

@media screen and (min-width: 768px) {
  .emotion-2 {
    display: block;
  }
}

@media screen and (min-width: 1024px) {
  .emotion-2 {
    display: none;
  }
}

@media screen and (min-width: 1440px) {
  .emotion-2 {
    display: none;
  }
}

.emotion-3 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-4 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.emotion-5 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
  fill: #2E2E2E;
  vertical-align: unset;
  display: inline-block;
}

.emotion-5.emotion-5 {
  width: 16px;
  height: 16px;
  fill: #2E2E2E;
}

.emotion-7 {
  font-family: "DM Sans",sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  letter-spacing: 0;
}

.emotion-8 {
  list-style-type: none;
  margin: 0px;
  padding: 0;
}

.emotion-8:last-child {
  margin-bottom: 24px;
}

<div
    class="emotion-0"
    data-testid="sidebar"
    open=""
    role="complementary"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-3"
        >
          <div
            class="emotion-4"
            data-testid="close-icon"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="emotion-5 emotion-6"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <nav
      aria-label="Sidebar"
      class="emotion-7"
      role="navigation"
    >
      <ul
        class="emotion-8"
        role="list"
      />
    </nav>
  </div>
  .emotion-0 {
  background-color: rgba(0, 0, 0, 0.8);
  position: fixed;
  overflow: auto;
  bottom: 0;
  top: 0;
  left: 0;
  right: 0;
  opacity: 1;
  visibility: visible;
  -webkit-transition: opacity 0.6s,right 0.3s;
  transition: opacity 0.6s,right 0.3s;
  z-index: 2;
}

@media screen and (min-width: 1024px) {
  .emotion-0 {
    display: none;
  }
}

<div
    class="emotion-0"
    data-testid="overlay"
  />
</DocumentFragment>
`;
